### PR TITLE
Add script for 3888 issue

### PR DESCRIPTION
**Pull request by GetmanetsIrina**
ATF Test Scripts to check #[FORDTCN-12994](https://luxproject.luxoft.com/jira/browse/FORDTCN-12994)

This PR is **ready for internal review** for review.

### Summary
Added new script to check resumption of CreateInteractionChoiceSet with and without VR command

### ATF version
develop

### Changelog
1. New script is added
2. Created script is added to the test set 8_2

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Review by VjKlepikov — APPROVED** [review]

**Review by VjKlepikov — CHANGES_REQUESTED** [review]
Please add test for resumption

**Comment by GetmanetsIrina** [comment]
Please find update in [d729878](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/173/commits/d729878e01b4f7f31c135b1d52dd3278fc921406)

File: test_scripts/Defects/8_2/3888_2_CreateInteractionChoiceSet_resumption_ignition_off.lua [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+for k, v in utils.spairs(testCases) do
+  runner.Title("Test case: CreateInteractionChoiceSet " .. k)
+  runner.Title("Preconditions")
+  runner.Step("Clean environment", common.preconditions)
+  runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+  runner.Step("Register App", common.registerAppWOPTU)
+  runner.Step("Activate App", common.activateApp)
+
+  runner.Step("App requests CreateInteractionChoiceSet " .. k, createInteractionChoiceSet, { v })
+  runner.Step("Unexpected disconnect", ignitionOff)
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888_2_CreateInteractionChoiceSet_resumption_ignition_off.lua]:
```suggestion
  runner.Step("Ignition off", ignitionOff)
```

**Review by VjKlepikov — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/8_2/3888_2_CreateInteractionChoiceSet_resumption_ignition_off.lua [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+-- Description: Check that SDL resumes CreatedInteractionChoice set with and without VR command
+--
+-- Precondition:
+-- 1. HMI and SDL are started
+-- 2. Mobile app is registered and activated
+--
+-- Test:
+-- 1. Mobile app adds CreateInteractionChoiceSet with and without VR command
+-- 2. Ignition off is performed
+-- 4. Ignition on is performed
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888_2_CreateInteractionChoiceSet_resumption_ignition_off.lua]:
```suggestion
-- 2. Ignition off is performed
-- 3. Ignition on is performed
```

**Review by VjKlepikov — CHANGES_REQUESTED** [review]
I propose to add resumption case for DeleteInteractionChoiceSet request and move common functions to common module

**Comment by GetmanetsIrina** [comment]
@VjKlepikov, please review

File: test_scripts/Defects/8_2/3888/common.lua [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+m.preconditions = actions.preconditions
+m.postconditions = actions.postconditions
+m.start = actions.start
+m.registerAppWOPTU = actions.registerAppWOPTU
+m.activateApp = actions.activateApp
+m.getMobileSession = actions.getMobileSession
+m.getHMIConnection = actions.getHMIConnection
+m.spairs = utils.spairs
+m.connectMobile = actions.mobile.connect
+m.getHMIAppId = actions.getHMIAppId
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/common.lua]: Should we use the mentioned proxy functions that are not in the scripts?

File: test_scripts/Defects/8_2/3888/common.lua [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered",
+    { appID = m.getHMIAppId(), unexpectedDisconnect = true })
+end
+
+function m.ignitionOff()
+  local isOnSDLCloseSent = false
+  m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "SUSPEND" })
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLPersistenceComplete")
+  :Do(function()
+    m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "IGNITION_OFF" })
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/common.lua]:
```suggestion
    m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "IGNITION_OFF" })
```
```suggestion
     m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "IGNITION_OFF" })
```

File: test_scripts/Defects/8_2/3888/3888_4_CreateInteractionChoiceSet_resumption_two_sets_ignition_off.lua [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+--[[ Test ]]
+common.Title("Preconditions")
+common.Step("Clean environment", common.preconditions)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+common.Step("Register App", common.registerAppWOPTU)
+common.Step("Activate App", common.activateApp)
+
+common.Step("CreateInteractionChoiceSet with vr command", common.createInteractionChoiceSet,
+  { common.testCases.with_vr_command })
+common.Step("CreateInteractionChoiceSet witout vr command", common.createInteractionChoiceSet,
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/3888_4_CreateInteractionChoiceSet_resumption_two_sets_ignition_off.lua]:
```suggestion
common.Step("CreateInteractionChoiceSet witout vr command", common.createInteractionChoiceSet,
```
```suggestion
common.Step("CreateInteractionChoiceSet without vr command", common.createInteractionChoiceSet,
```

Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/3888_4_CreateInteractionChoiceSet_resumption_two_sets_ignition_off.lua:31]:
```suggestion
common.Step("CreateInteractionChoiceSet witout vr command", common.createInteractionChoiceSet,
```
```suggestion
common.Step("CreateInteractionChoiceSet without vr command", common.createInteractionChoiceSet,
```

File: test_scripts/Defects/8_2/3888/3888_6_DeleteInteractionChoiceSet_during_resumption_unexpected_disc.lua [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+-- 3. Unexpected disconnect is performed
+-- 4. Mobile app requests RAI with actual hashId
+-- SDL does:
+--  - restore UI part of CreateInteractionChoiceSet internally and
+--    requests VR.AddCommand("type":"Choice") only in case CreateInteractionChoiceSet was added with VR command
+-- 5. HMI responds with SUCCESS to VR.AddCommands and with erroneous result code to AddSubMenu
+-- SDL does:
+--  - remove UI part of CreateInteractionChoiceSet internally and requests VR.DeleteCommand only for choice set that
+--     was added with VR command
+--  - send RAI response with "SUCCESS" result code after resumption data is succeeded
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/3888_6_DeleteInteractionChoiceSet_during_resumption_unexpected_disc.lua:23]:
```suggestion
--  - send RAI response with "SUCCESS" result code after resumption data is succeeded
```
```suggestion
--  - send RAI response with "RESUME_FAILED" result code after resumption data is succeeded
```

Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/3888_6_DeleteInteractionChoiceSet_during_resumption_unexpected_disc.lua]:
```suggestion
--  - send RAI response with "SUCCESS" result code after resumption data is succeeded
```
```suggestion
--  - send RAI response with "RESUME_FAILED" result code
```

Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/3888_6_DeleteInteractionChoiceSet_during_resumption_unexpected_disc.lua:23]:
```suggestion
--  - send RAI response with "SUCCESS" result code after resumption data is succeeded
```
```suggestion
--  - send RAI response with "RESUME_FAILED" result code
```

File: test_scripts/Defects/8_2/3888/3888_7_DeleteInteractionChoiceSet_during_resumption_ignition_off.lua [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+--
+-- Test:
+-- 1. Mobile app adds CreateInteractionChoiceSet with and without VR command
+-- 2. Ignition OFF and ON are performed
+-- 3. Mobile app requests RAI with actual hashId
+-- SDL does:
+--  - restore UI part of CreateInteractionChoiceSet internally and
+--    requests VR.AddCommand("type":"Choice") only in case CreateInteractionChoiceSet was added with VR command
+--  - send RAI response with "SUCCESS" result code after resumption data is succeeded
+---------------------------------------------------------------------------------------------------
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/3888_7_DeleteInteractionChoiceSet_during_resumption_ignition_off.lua]: Please update the description according to the steps

Comment (VjKlepikov) [inline test_scripts/Defects/8_2/3888/3888_7_DeleteInteractionChoiceSet_during_resumption_ignition_off.lua:18]: Please update the description according to the steps

**Review by VjKlepikov — CHANGES_REQUESTED** [review]

**Comment by GetmanetsIrina** [comment]
@VjKlepikov, please find update in [8f758fb](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/173/commits/8f758fb0f9fbaa9591b9175b40d86269a3058113)

**Review by VjKlepikov — APPROVED** [review]

**Comment by GetmanetsIrina** [comment]
moved to open

**Pull request closed without merging**